### DISCUSSION TITLE
Fish lsp and/or linter recommendations?

### DISCUSSION THREAD

**Discussion by rickswe** [fish-shell/fish-shell#10428] [Q&A] [answered]
Hello,

I've looked around for an lsp but it looks like are not many popular ones out there.
For example in bash there's bash langauge server and shellcheck which are really popular.

I'm in the process of moving my bashrc and profile to config.fish.

I am getting some linting but i'm not sure where it's comming from:

```sh
set -x XDG_DATA_HOME "${HOME}/.local/share"
```

```text
Variables cannot be bracketed. In fish, please use "$HOME"
```

I don't think this is shellcheck or bashls.

Thanks

**Comment by faho** [comment, marked as answer]
> I've looked around for an lsp but it looks like are not many popular ones out there.

I am unaware of an lsp server or linter for fish.

> I am getting some linting but i'm not sure where it's comming from:

That's not linting, that's a syntax error, coming directly from fish.

**Reply by rickswe** [reply]
>That's not linting, that's a syntax error, coming directly from fish.

I'm not sure what it is. I just moved from my vanilla config to LazyVim and it installs a lot of stuff out of the box.
This is the trouble quickfix window:

![fish-lsp](https://github.com/fish-shell/fish-shell/assets/76216665/fecb0ca0-27a2-4ef3-bc41-ee4eb53b0fac)

**Reply by faho** [reply]
Yes, that's a syntax error. Fish is not posix-compatible, it does not use `if ...; then; ...; fi`, it uses `if ...; ...; end` - `fi` is not special to fish at all, so this if-block is, in fact, incomplete, rendering the file broken.

It appears your vim config sends the file off to fish for syntax checking - `fish --no-execute` will do that.

**Reply by rickswe** [reply]
>Fish is not posix-compatible, it does not use if ...; then; ...; fi

I understand this is not valid fish syntax, I created an error to demonstrate the error messages. 

>It appears your vim config sends the file off to fish for syntax checking 

I'm not sure what it is. I'll eventually find out once I go back to messing with my nvim config.
I already took out bash-language-server, shellcheck, and checkmate. Mason does not
show them installed and LspInfo does not show anything. No big deal.
